platform/linuxbsd/detect.py

```diff
@@ -111,15 +112,24 @@ def configure(env):
             env["CXX"] = "clang++"
         env.extra_suffix = ".llvm" + env.extra_suffix
 
-    if env["use_lld"]:
-        if env["use_llvm"]:
-            env.Append(LINKFLAGS=["-fuse-ld=lld"])
-            if env["use_thinlto"]:
-                # A convenience so you don't need to write use_lto too when using SCons
-                env["use_lto"] = True
+    if env["linker"] != "default":
+        print("Using linker program: " + env["linker"])
+        if env["linker"] == "mold" and using_gcc(env):  # GCC < 12.1 doesn't support -fuse-ld=mold.
+            cc_version = get_compiler_version(env)
+            cc_semver = (int(cc_version["major"]), int(cc_version["minor"]))
+            if cc_semver < (12, 1):
+                env.Append(LINKFLAGS=["-B/usr/libexec/mold"])
```

Comment:
Seems like an arbitrary location, maybe it should use output of `which mold` or something like this, to ensure it will work on any distro.

Reply:
It is indeed somewhat arbitrary but this should be the default install path for a distro-packaged mold.

We can't use `which mold` as this gives `/usr/bin/mold`, but for the `-B` linker option GCC expects a path to a folder with a file named `ld` (yes that's software designed in the 90s...):
```
$ ls -l /usr/libexec/mold/
total 0
lrwxrwxrwx 1 root root 14 Jul  1 10:45 ld -> ../../bin/mold*
```

Expand "A classic way to use mold" in https://github.com/rui314/mold#how-to-use for details.

Possible install locations for this would be:
- `/usr/libexec/mold` if using `PREFIX=/usr`, which is what most Linux distros would do when packaging
- `/usr/local/libexec/mold` if using `PREFIX=/usr/local`, which is the default and hence what users who self-install mold from source might have
- Any arbitrary location if users pass a different `PREFIX` or hack the `Makefile`

I chose to only support the distro package scenario as the rest is IMO too difficult to handle. We could maybe at least do a check that `/usr/libexec/mold/ld` or `/usr/local/libexec/mold/ld` exist and use it, or abort.

Reply:
OK, but a check for the `/usr/local/..` is probably good to have.

Reply:
Yeah I agree, see last push.